### ISSUE TITLE
SSE format is not matching the standard and official one.

### ISSUE THREAD

**Issue by kmruiz** [vlingo/vlingo-http#28]
The current SSE format is using some conventions only understood by vlingo itself. It means that you can't use vlingo SSE to feed a browser with events. If we want to widden the usage of SSE, it should be compliant with the standards:

https://developer.mozilla.org/en-US/docs/Web/API/Server-sent_events/Using_server-sent_events

**Comment by VaughnVernon** [comment]
@kmruiz It is not meant to be different. (What is it?) Please fix.

**Comment by kmruiz** [comment]
In [SseClient](https://github.com/vlingo/vlingo-http/blob/master/src/main/java/io/vlingo/http/resource/sse/SseClient.java) there are some issues (but I don't know if there are more):

According to the spec:

* Headers can only be sent once (we are sending headers multiple times)
* Headers must be sent when the connection is open (at least the Content-Type)

However if you see the send method, all of them are sending header information, and the mandatory headers are sent only with the first event.

If this is the only issue, it should be quite trivial to fix.

**Comment by VaughnVernon** [comment]
@kmruiz Sure. Since you are already on this can you make the changes?

**Comment by VaughnVernon** [comment]
@kmruiz I am not sure I understand whether the problem is from the client side or the server side. Would you mind providing the context/perspective including the current incorrect header information and an example of the correct header usage? It would help me be helpful.

**Comment by kmruiz** [comment]
How SSE works is as following (if I understand it right):

I am a client that connects to a SSE endpoint, so I will receive the following output from the server side:

```200 OK
Content-Type: text/event-stream
Transfer-Encoding: chunked
```
The client understands the content type and the transfer encoding, so will be keeping the connection open for events.

When the server has an event (or a set of events), it needs to send a chunked answer. A chunk has 2 parts:

1. The chunk size (in hexadecimal format)
2. The chunk itself

Those are split by newlines, so for example, an event would be like this:

`11\nevent:hello world\n`

Right now, what vlingo is doing, is generating a complete HTTP response on each set of events, for example:

https://github.com/vlingo/vlingo-http/blob/master/src/main/java/io/vlingo/http/resource/sse/SseClient.java#L81

On how to reproduce the issue, you can start the backend service in examples, and execute a curl -v to the /vault/ endpoint. You will see an error saying that the answer is incorrect.

**Comment by VaughnVernon** [comment]
@kmruiz Ok, now I follow. TBF I found the spec very sketchy in this case and any number of examples I found were unclear. I can only assume that you are correct. I would like to have the opinion of Aleix here.

**Comment by VaughnVernon** [comment]
@aleixyz You could make this (hopefully) simple change to bring vlingo-http SSE into spec. Currently I output an HTTP response header with every event(s) packet push. Instead there should be a one-time response header that indicates the that the channel is open. 

It is difficult to find full examples of this working, and the tutorials are vague on the actual response formats and event pushes. I found this example showing a presumable correct protocol:

https://github.com/mariomac/jeasse/blob/master/jeasse-common/src/main/java/info/macias/sse/EventTarget.java

https://github.com/mariomac/jeasse/blob/master/jeasse-servlet3/src/main/java/info/macias/sse/SseDispatcher.java

According to the above, this is the valid one-time response with header:

```
HTTP/1.1 200 OK
Content-type: text/event-stream;charset=utf-8
Cache-Control: no-cache
Connection: keep-alive
```

The rest of the pushes are just events. If the server closes the channel/connection it doesn't send any responses/data; it just closes.

**Comment by VaughnVernon** [comment]
Marking this fixed as of 0.8.7.
